### PR TITLE
Document DynamoDB backend billing_mode option

### PR DESCRIPTION
**Pull request by lxea**
part of #27674

### CODE REVIEW

File: docs/pages/reference/backends.mdx [hunk 1]
```diff
@@ -486,6 +486,10 @@ teleport:
     # By default, Teleport stores audit events with an AWS TTL of 1 year.
     # This value can be configured as shown below. If set to 0 seconds, TTL is disabled.
     audit_retention_period: 365d
+
+    # Enables On-Demand billing for the dynamodb table, if set to false, provisioned mode will be used.
```
Comment (zmb3) [inline docs/pages/reference/backends.mdx:490]:
```suggestion
    # Enables On-Demand billing for the DynamoDB table.
    # Set to false to use provisioned capacity.
```

As an aside, defaulting a boolean to true is awkward and flies against the "make the zero value useful" paradigm.

If we want to start defaulting to on demand billing, I think we could make this clearer by either:

1. Inverting this, so you have to opt in to provisioned mode with something like `provisioned_billing: true`. (This also makes it more obvious that the capacity fields are ignored if this is not set.
2. Getting rid of booleans completely, and having `billing_mode: provisioned` or `billing_mode: on_demand`.

File: docs/pages/reference/backends.mdx [hunk 2]
```diff
@@ -556,6 +560,9 @@ teleport:
     write_min_capacity: int
     write_max_capacity: int
     write_target_value: float
+    # Enables On-Demand billing
+    # default: true
+    on_demand: [true|false]
```
Comment (zmb3) [inline docs/pages/reference/backends.mdx:565]: I would move this above the read/write capacity values and also mention here that the capacity values are ignored for on-demand mode.

File: docs/pages/reference/backends.mdx [hunk 3]
```diff
@@ -486,6 +486,10 @@ teleport:
     # By default, Teleport stores audit events with an AWS TTL of 1 year.
     # This value can be configured as shown below. If set to 0 seconds, TTL is disabled.
     audit_retention_period: 365d
+
+    # Enables On-Demand billing for the dynamodb table, if set to false, provisioned mode will be used.
+    # default: true
+    on_demand: true
```
Comment (zmb3) [inline docs/pages/reference/backends.mdx:492]:
```suggestion
    on_demand_billing: true
```

File: docs/pages/reference/backends.mdx [hunk 4]
```diff
@@ -556,6 +560,9 @@ teleport:
     write_min_capacity: int
     write_max_capacity: int
     write_target_value: float
+    # Enables On-Demand billing
```
Comment (zmb3) [inline docs/pages/reference/backends.mdx:563]: These settings only take effect if the table does not exist and Teleport has to create it, right?

If so, that's not clear and we should probably clarify.

File: docs/pages/reference/backends.mdx (hunk 1 above)
Comment (r0mant) [reply docs/pages/reference/backends.mdx:490]: Right, I agree "billing_mode" enum sounds like a better fit than a boolean constant. @lxea Shall we update your PR while it's still not merged? https://github.com/gravitational/teleport/pull/27674

**Comment by alexfornuto** [comment]
Waiting for edits based on Zac's comments before I perform a copy review.

**Review by alexfornuto — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Comment by r0mant** [comment]
I've added `do-not-merge` label on this one so we don't accidentally merge it until the corresponding [implementation PR](https://github.com/gravitational/teleport/pull/27674) is merged.

**Pull request merged**